src/python/binary_flash.py

```diff
@@ -62,7 +62,7 @@ def upload_stm32_stlink(args, options: FirmwareOptions):
     app_start = flash_start + offset
 
     if options.bootloader is not None:
-        stlink.program_flash('bootloader/' + options.bootloader, flash_start, erase=True, verify=True, initialize_comms=True)
+        stlink.program_flash(args.fdir + '/bootloader/' + options.bootloader, flash_start, erase=True, verify=True, initialize_comms=True)
```

Comment:
Should really calculate a local variable as `args.fdir` may be set to `None`, this is the pattern used in `binary_configurator.py:378`

```python
firmware_dir = '' if args.fdir is None else args.fdir + '/'
stlink.program_flash(firmware_dir + 'bootloader/' + options.bootloader, flash_start, erase=True, verify=True, initialize_comms=True)
```

Reply:
Thanks for the feedback. How does it look now?

Reply:
Looks good. I'll give a test run this evening.